codegen-client/src/main/kotlin/software/amazon/smithy/rust/codegen/client/smithy/generators/config/ServiceConfigGenerator.kt

```diff
@@ -412,6 +415,27 @@ class ServiceConfigGenerator(
                 it.section(ServiceConfig.BuilderImpl)(this)
             }
 
+            if (runtimeMode.defaultToOrchestrator) {
+                rustTemplate(
+                    """
+                    /// Adds a runtime plugin to the config.
+                    ##[doc(hidden)]
+                    pub fn runtime_plugin(mut self, plugin: impl #{RuntimePlugin} + 'static) -> Self {
+                        self.runtime_plugins.push(#{SharedRuntimePlugin}::new(plugin));
+                        self
+                    }
+
+                    /// Adds a runtime plugin to the config.
+                    ##[doc(hidden)]
+                    pub fn push_runtime_plugin(&mut self, plugin: #{SharedRuntimePlugin}) -> &mut Self {
```

Comment:
I think this can be pub(crate)

Reply:
I seem to be running into a compiler bug where rustc erroneously gives a dead code warning even with `#[allow(unused)]` or `#[allow(dead_code)]` on top of the function. Going to stick with the pub/doc-hidden combo for now.

Reply:
Changed to `pub(crate)` in cacf7bac9993f8785ba087aa185e4fb69d5c1419